.make/test.mk

```diff
@@ -0,0 +1,381 @@
+# Normal test targets
+#
+# These test targets are the ones that will be invoked from the outside. If
+# they are called and the artifacts already exist, then the artifacts will
+# first be cleaned and recreated. This ensures that the tests are always
+# executed.
+#-------------------------------------------------------------------------------
+
+.PHONY: test-all
+## Runs test-unit and test-integration targets.
+test-all: prebuild-check test-unit
+
+.PHONY: test-unit
+## Runs the unit tests and produces coverage files for each package.
+test-unit: prebuild-check clean-coverage-unit $(COV_PATH_UNIT)
+
+.PHONY: test-unit-no-coverage
+## Runs the unit tests and WITHOUT producing coverage files for each package.
+test-unit-no-coverage: prebuild-check $(SOURCES)
+	$(call log-info,"Running test: $@")
+	$(eval TEST_PACKAGES:=$(shell go list ./... | grep -v $(ALL_PKGS_EXCLUDE_PATTERN)))
+	F8_DEVELOPER_MODE_ENABLED=1 F8_RESOURCE_UNIT_TEST=1 F8_LOG_LEVEL=$(F8_LOG_LEVEL) go test -vet off $(GO_TEST_VERBOSITY_FLAG) $(TEST_PACKAGES)
+
+# Starts the WIT server and waits until its running
+# define start-wit
+# 	echo "Starting WIT and ensuring that it is running..."; \
+# 	F8_LOG_LEVEL=ERROR ./wit+pmcd.sh & 
+# 	while ! nc -z localhost 8080 < /dev/null; do \
+# 		printf .; \
+# 		sleep 5 ; \
+# 	done; \
+# 	echo "WIT is RUNNING!";
+# endef
+
+# Downloads docker-compose to tmp/docker-compose if it does not already exist.
+define download-docker-compose
+	@if [ ! -f "$(DOCKER_COMPOSE_BIN_ALT)" ]; then \
+		echo "Downloading docker-compose to $(DOCKER_COMPOSE_BIN_ALT)"; \
+		UNAME_S=$(shell uname -s); \
+		UNAME_M=$(shell uname -m); \
+		URL="https://github.com/docker/compose/releases/download/1.22.0/docker-compose-$${UNAME_S}-$${UNAME_M}"; \
+		curl --silent -L $${URL} > $(DOCKER_COMPOSE_BIN_ALT); \
+		chmod +x $(DOCKER_COMPOSE_BIN_ALT); \
+	fi
+endef
+
+#-------------------------------------------------------------------------------
+# Inspect coverage of unit tests, integration or remote tests in either pure
+# console mode or in a browser (*-html).
+#
+# If the test coverage files to be evaluated already exist, then no new
+# tests are executed. If they don't exist, we first run the tests.
+#-------------------------------------------------------------------------------
+
+# Prints the total coverage of a given package.
+# The total coverage is printed as the last argument in the
+# output of "go tool cover". If the requested test name (first argument)
+# Is *, then unit, integration and remote tests will be combined
+define print-package-coverage
+$(eval TEST_NAME:=$(1))
+$(eval PACKAGE_NAME:=$(2))
+$(eval COV_FILE:="$(COV_DIR)/$(PACKAGE_NAME)/coverage.$(TEST_NAME).mode-$(COVERAGE_MODE)")
+ @if [ "$(TEST_NAME)" == "*" ]; then \
+  UNIT_FILE=`echo $(COV_FILE) | sed 's/*/unit/'`; \
+  INTEGRATON_FILE=`echo $(COV_FILE) | sed 's/*/integration/'`; \
+  REMOTE_FILE=`echo $(COV_FILE) | sed 's/*/remote/'`; \
```

Comment:
you probably don't need the `REMOTE_FILE`

Reply:
@xcoulon I don't think so, we should modified the logic defined within this function. It incur more work/issue without real benefit.  I really don't think we should touch this function.

Reply:
but why would we keep something we don't need? I remove it in other repos and it wasn't so much work AFAIR

Reply:
I don't see any problem in keeping this as it is.  If you have sill have problem then only I will remove it.